src/check/arbitrary/ArrayArbitrary.ts

```diff
@@ -84,25 +89,27 @@ export class ArrayArbitrary<T> extends Arbitrary<T[]> {
         numSkippedInRow += 1;
       }
     }
-    return this.wrapper(items, false);
+    return this.wrapper(items, undefined);
   }
-  private shrinkImpl(items: Shrinkable<T>[], shrunkOnce: boolean): Stream<Shrinkable<T>[]> {
+  private shrinkImpl(items: Shrinkable<T>[], lengthArbContext: unknown): Stream<[Shrinkable<T>[], unknown]> {
     if (items.length === 0) {
-      return Stream.nil<Shrinkable<T>[]>();
+      return Stream.nil<[Shrinkable<T>[], unknown]>();
     }
-    const size = this.lengthArb.shrinkableFor(items.length, shrunkOnce);
-    return size
-      .shrink()
-      .map((l) => items.slice(items.length - l.value))
-      .join(items[0].shrink().map((v) => [v].concat(items.slice(1))))
+    return this.lengthArb
+      .contextualShrink(items.length, lengthArbContext)
+      .map(
+        ([l, nextLengthArbContext]) =>
+          [items.slice(items.length - l), nextLengthArbContext] as [Shrinkable<T>[], unknown]
+      )
+      .join(items[0].shrink().map((v) => [[v].concat(items.slice(1)), lengthArbContext]))
```

Comment:
We introduced a change here compared to previous implementation ⚠️ 

In the past, wrapper was doing:
```js
return new Shrinkable(vs, () => this.shrinkImpl(items, shrunkOnce).map((v) => this.wrapper(v, true)))
```

So `shrunkOnce` was truthy on the outputs of `shrinkImpl`. Even for the line: `      .join(items[0].shrink().map((v) => [v].concat(items.slice(1))))`. Now as we pass the previous context here, we implicitly say: `shrunkOnce` is false which is incorrect.